variant_concordance/var_conc_vcf.wdl

```diff
@@ -0,0 +1,223 @@
+            with open(sample_pairs, 'rt') as f:
+                pairs = [[line.strip().split('\t')[0], line.strip().split('\t')[1]] for line in f]
+            eprint(f'{len(pairs)} sample pairs read')
+            sample_index = read_sample_index(vcf)
+            eprint(f'{len(sample_index)-9} samples in vcf')
+
+            for pair in pairs:
+                if pair[0] not in sample_index:
+                    eprint(f'sample {pair[0]} not found in vcf, quit')
+                    quit(3)
+                if pair[1] not in sample_index:
+                    eprint(f'sample {pair[1]} not found in vcf, quit')
+                    quit(3)
+
+            with gzip.open(vcf, 'rt') as f:
+                line = f.readline().strip()
+                while line.startswith('##'):
+                    line = f.readline().strip()
+                print('\t'.join(['variant', 'af0', 'miss0', 'miss1',
+                                 'ref0_ref1', 'ref0_het1', 'ref0_alt1',
+                                 'het0_ref1', 'het0_het1', 'het0_alt1',
+                                 'alt0_ref1', 'alt0_het1', 'alt0_alt1',
+                                 'het0_het1_prop', 'hom0_hom1_prop']))
+                eprint('printing counts to stdout')
+                for line in f:
+                    s = line.strip().split('\t')
+                    # counts are: [0 ref_ref, 1 ref_het, 2 ref_alt, 3 het_ref, 4 het_het, 5 het_alt, 6 alt_ref, 7 alt_het, 8 alt_alt]
+                    counts = [0 for i in range(0,9)]
+                    miss0=miss1=0
+                    for pair in pairs:
+                        all0 = count_alt_alleles(s[sample_index[pair[0]]])
+                        all1 = count_alt_alleles(s[sample_index[pair[1]]])
+                        if all0 == -1:
+                            miss0 = miss0 + 1
+                        if all1 == -1:
+                            miss1 = miss1 + 1
+                        if all0 == -1 or all1 == -1:
+                            continue
+                        count_index = 3 * all0 + all1
+                        counts[count_index] = counts[count_index] + 1
+
+                    het_total = counts[1]+counts[3]+counts[4]+counts[5]+counts[7]
+                    het_prop = counts[4]/het_total if het_total > 0 else -1
+
+                    # calculate hom_total and hom_prop for decide minor allele
+                    n_alt0 = 2*sum(counts[6:9]) + sum(counts[3:6])
+                    n_ref0 = 2*len(pairs) - n_alt0
+                    if n_alt0 < n_ref0:
+                        hom_total = counts[2]+counts[5]+counts[6]+counts[7]+counts[8]
+                        hom_prop = counts[8]/hom_total if hom_total > 0 else -1
+                    else:
+                        hom_total = counts[0]+counts[1]+counts[2]+counts[3]+counts[6]
+                        hom_prop = counts[0]/hom_total if hom_total > 0 else -1
+
+                    af = n_alt0/2/len(pairs)
```

Comment:
missing values should be deducted from here also.

Reply:
👍🏼